content/blog/writing-crossguard-policies/index.md

```diff
@@ -0,0 +1,167 @@
+
+```typescript
+const elbLoggingEnabled: ResourceValidationPolicy = {
+    name: "elb-logging-enabled",
+    description: "Checks whether the Application Load Balancers and the Classic Load Balancers have logging enabled.",
+    enforcementLevel: "mandatory",
+    validateResource: [
+        validateTypedResource(aws.elasticloadbalancing.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Elastic Load Balancer must have access logs enabled.");
+            }
+        }),
+        validateTypedResource(aws.elasticloadbalancingv2.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Elastic Load Balancer must have access logs enabled.");
+            }
+        }),
+        validateTypedResource(aws.applicationloadbalancing.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Application Load Balancer must have access logs enabled.");
+            }
+        }),
+    ],
+};
+```
+
+## Stack Validation Policies
+
+Policies of `StackValidationPolicy` are run against the stack as a whole. These policies are run _after_ all stack resources are registered and thus do not block an out-of-compliance resource from being created or modified, but do fail the preview or update. To avoid creating out-of-compliance resources, we recommend running a preview command before an update. However, there are cases where a policy requires a resource to exist in order to perform its validation, such as inspecting the resource's output properties that are only available to already provisioned resources (e.g. checking if a resource's `id` is referenced by another resource). In such cases, the policy can skip reporting violations during previews and only report violations during updates when the information it needs is available. Thus, the preview will succeed and the resources will be created or modified during the update, but the overall update operation will fail if any resources are in violation.
+
+A stack validation function is passed `args` with information about all of the resources in the stack (such as each resource's type, _output_ properties, name, and URN) and a `reportViolation` callback that can be called to report a policy violation. `reportViolation` requires passing a `message` and also supports an optional `urn` parameter, to specify the URN of the resource in violation (otherwise violation is associated with the stack itself).
```

Comment:
Nit: The `urn` of a resource is probably unknown to the reader... maybe it's just confusing and we should cut it? Or alternatively, link to somewhere in the docs where we explain what it is?